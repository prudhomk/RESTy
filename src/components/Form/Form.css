.Form {
    margin-right: 0.3vw;
    width: 81.9vw;
    display: flex;
    flex-direction: column;
}
.Form section {
    display: flex;
    width: 100%;
    justify-content: flex-start;
    align-items: center;
    flex-direction: row;
}
.Form section input[type="radio"] {
    visibility: hidden;
    height: 0;
    width: 0;
    margin-left: 0;
}
.Form section label {
    padding: 4px 16px;
    cursor: pointer;
}
.Form section input[type="radio"]:checked + label {
    background-color: #d0803b;
    color: white;
}
.Form input {
    margin-bottom: 8px;
    align-self: flex-start;
}

.Form button {
    margin-left: 2vw;
}

.Form textarea {
    margin-top: 5px;
    width: 50%;
    height: 6vw;
    align-self: flex-start;
}
.url {
    width: 100%;
    height: 25px;
    font-size: 12px;
    margin-top: 5px;
}
