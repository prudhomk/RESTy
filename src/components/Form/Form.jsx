/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import RadioButtons from '../Radio/RadioButtons';
import styles from './Form.css';


const Form = ({ url, body, method, onSubmit, onChange }) => {
  return (
    <section className={styles.Form}>
      <form onSubmit={onSubmit}>
        <input type="text" name="url" placeholder="URL" value={url} onChange={onChange} />
        <section>
          <RadioButtons method={method} onChange={onChange}/>
          <button>Go!</button>
        </section>
        <textarea placeholder="Raw JSON Body" name="body" value={body} onChange={onChange}></textarea>
      </form>
    </section>
  );
};

Form.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Form;
