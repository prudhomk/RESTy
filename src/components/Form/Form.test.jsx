import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Form from '../Form/Form';

describe('Form component', () => {
  afterEach(() => cleanup());
  it('renders Form', () => {
    const { asFragment } = render(<Form />);
    expect(asFragment()).toMatchSnapshot();
  });
});
