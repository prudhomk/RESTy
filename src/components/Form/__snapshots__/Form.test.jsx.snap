// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<DocumentFragment>
  <section
    class="Form"
  >
    <form>
      <input
        name="url"
        placeholder="URL"
        type="text"
        value=""
      />
      <section>
        <div>
          <input
            id="get"
            name="method"
            type="radio"
            value="GET"
          />
          <label
            for="get"
          >
            GET
          </label>
          <input
            id="post"
            name="method"
            type="radio"
            value="POST"
          />
          <label
            for="post"
          >
            POST
          </label>
          <input
            id="put"
            name="method"
            type="radio"
            value="PUT"
          />
          <label
            for="put"
          >
            PUT
          </label>
          <input
            id="patch"
            name="method"
            type="radio"
            value="PATCH"
          />
          <label
            for="patch"
          >
            PATCH
          </label>
          <input
            id="delete"
            name="method"
            type="radio"
            value="DELETE"
          />
          <label
            for="delete"
          >
            DELETE
          </label>
        </div>
        <button>
          Go!
        </button>
      </section>
      <textarea
        name="body"
        placeholder="Raw JSON Body"
      />
    </form>
  </section>
</DocumentFragment>
`;
